src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -472,6 +472,48 @@ impl LogicalAggBuilder {
         }
         None
     }
+
+    pub fn syntax_check(&self) -> Result<()> {
```

Comment:
IMHO it's better to add to TODO here to remind that someday we might disable the Expand optimization and let the backend to dedup inputs when users do this kind of queries, cuz we **don't have to** ban these queries if it doesn't require too many code to implement DISTINCT in backend?

Reply:
Also since we can do the syntax check with global sight to all agg calls here, the following check happens in each `bind_function` can be removed:

https://github.com/singularity-data/risingwave/blob/e7ce1ba7abe44b76c4bc3699dd2effe66522422d/src/frontend/src/binder/expr/function.rs#L118-L124